x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/mapper/AggregateMetricFieldMapper.java

```diff
@@ -0,0 +1,340 @@
+                            parsedMetrics.add(metricsStr[i]);
+                        }
+                    }
+                    builder.metrics(parsedMetrics);
+                    iterator.remove();
+                } else if (propName.equals(Names.IGNORE_MALFORMED)) {
+                    builder.ignoreMalformed(XContentMapValues.nodeBooleanValue(propNode, name + "." + Names.IGNORE_MALFORMED));
+                    iterator.remove();
+                } else if (TypeParsers.parseMultiField(builder, name, parserContext, propName, propNode)) {
+                    iterator.remove();
+                }
+            }
+            return builder;
+        }
+    }
+
+    public static final class AggregateMetricFieldType extends MappedFieldType {
+
+        public AggregateMetricFieldType() {
+        }
+
+        AggregateMetricFieldType(AggregateMetricFieldType other) {
+            super(other);
+        }
+
+        @Override
+        public MappedFieldType clone() {
+            return new AggregateMetricFieldType(this);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            if (hasDocValues() == true) {
+                return new DocValuesFieldExistsQuery(name());
+            } else {
+                throw new QueryShardException(context, "field  " + name() + " of type [" + CONTENT_TYPE + "] " +
+                    "has no doc values and cannot be searched");
+            }
+        }
+
+        @Override
+        public Query termQuery(Object value, @Nullable QueryShardContext context) {
+            throw new QueryShardException(context, "[" + CONTENT_TYPE + "] field do not support searching, " +
+                "use dedicated aggregations instead: ["
+                + name() + "]");
+        }
+    }
+
+    private Explicit<Boolean> ignoreMalformed;
+
+    private Explicit<Set<String>> metrics;
+
+    private AggregateMetricFieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType,
+        Settings indexSettings, MultiFields multiFields, Explicit<Boolean> ignoreMalformed, Explicit<Set<String>> metrics,
+        CopyTo copyTo) {
+        super(simpleName, fieldType, defaultFieldType, indexSettings, multiFields, copyTo);
+        this.ignoreMalformed = ignoreMalformed;
+        this.metrics = metrics;
+    }
+
+    @Override
+    public AggregateMetricFieldType fieldType() {
+        return (AggregateMetricFieldType) super.fieldType();
+    }
+
+    @Override
+    protected String contentType() {
+        return fieldType.typeName();
+    }
+
+    @Override
+    protected AggregateMetricFieldMapper clone() {
+        return (AggregateMetricFieldMapper) super.clone();
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context, List<IndexableField> fields) throws IOException {
+        Object metricsAsObject;
```

Comment:
I think this logic should in the parse() method not here? This method should never be called.

Reply:
By checking code at [`FieldMapper#parse()` ](https://github.com/elastic/elasticsearch/blob/f7a5b739729bb00c221d0ad54f0c5062876640da/server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java#L277) it seems that `parse()` is calling `parseCreateField()` and only handles any exceptions thrown().  

Finally it calls `multiFields.parse(this, context);` which does nothing our case. I do not really understand why overriding `parse()` is preferable to `parseCreateField()`. Happy to change this though.

Reply:
++ to use `parseCreateField`. We parse the content from the `_source` so I don't see why it should never be called @iverase ?

Reply:
I must say I am not sure but so far has always been my choice. I will check.